meshroom/core/plugins.py

```diff
@@ -46,16 +47,123 @@ def validateNodeDesc(nodeDesc: desc.Node) -> list:
     return errors
 
 
+class ProcessEnvType(Enum):
+    """ Supported process environments. """
+    DIRTREE = "dirtree",
+    REZ = "rez"
+
+
 class ProcessEnv(BaseObject):
     """
     Describes the environment required by a node's process.
+
+    Args:
+        folder: the source folder for the process.
+        envType: (optional) the type of process environment.
+        uri: (optional) the Unique Resource Identifier to activate the environment.
     """
 
-    def __init__(self, folder: str):
+    def __init__(self, folder: str, envType: ProcessEnvType = ProcessEnvType.DIRTREE, uri: str = ""):
         super().__init__()
-        self.binPaths: list = [Path(folder, "bin")]
-        self.libPaths: list = [Path(folder, "lib"), Path(folder, "lib64")]
-        self.pythonPathFolders: list = [Path(folder)] + self.binPaths
+        self._folder: str = folder
+        self._processEnvType: ProcessEnvType = envType
+        self.uri: str = uri
+
+    def getEnvDict(self) -> dict:
+        """ Return the environment dictionary if it has been modified, None otherwise. """
+        return None
+
+    def getCommandPrefix(self) -> str:
+        """ Return the prefix to the command line that will be executed by the process. """
+        return ""
+
+    def getCommandSuffix(self) -> str:
+        """ Return the suffix to the command line that will be executed by the process. """
+        return ""
+
+
+class DirTreeProcessEnv(ProcessEnv):
+    """
+    """
+    def __init__(self, folder: str):
+        super().__init__(folder, ProcessEnvType.DIRTREE)
+
+        self.binPaths: list = [str(Path(folder, "bin"))]
+        self.libPaths: list = [str(Path(folder, "lib")), str(Path(folder, "lib64"))]
+        self.pythonPaths: list = [str(Path(folder))] + self.binPaths
+
+    def getEnvDict(self) -> dict:
+        env = os.environ.copy()
+        env["PYTHONPATH"] = f"{_MESHROOM_ROOT}{os.pathsep}{os.pathsep.join(self.pythonPaths)}{os.pathsep}{os.getenv('PYTHONPATH', '')}"
```

Comment:
This looks easier to read to me (avoiding the repetition of pathsep):
```suggestion
        env["PYTHONPATH"] = os.pathsep.join([f"{_MESHROOM_ROOT}"] + self.pythonPaths + [os.getenv('PYTHONPATH', '')])
```